src/luxonis_ml/data/dataset.py

```diff
@@ -196,16 +102,22 @@ def __init__(
                 "Must provide either dataset_name or dataset_id when initializing LuxonisDataset"
             )
 
-        if team_id is None:
-            team_id = os.getenv("LUXONISML_TEAM_ID", "offline")
-        if team_name is None:
-            team_name = os.getenv("LUXONISML_TEAM_ID", "offline")
-
-        self.conn = foo.get_db_conn()
-
         self.base_path = os.getenv(
             "LUXONISML_BASE_PATH", str(Path.home() / "luxonis_ml")
         )
+        credentials_cache_file = str(Path(self.base_path) / "credentials.json")
+        if os.path.exists(credentials_cache_file):
+            with open(credentials_cache_file) as file:
+                self.config = json.load(file)
+        else:
+            self.config = {}
+
+        if team_id is None:
+            team_id = self._get_config("LUXONISML_TEAM_ID", "offline")
+        if team_name is None:
+            team_name = self._get_config("LUXONISML_TEAM_ID", "offline")
+
+        self.bucket = self._get_config("LUXONISML_BUCKET", None)
```

Comment:
Does this come with the provider as well? (GCS, S3, ...)

Reply:
The provider is currently specified when initializing `LuxonisDataset`. Do you think it would make more sense to come from an environment variable?